core/src/main/java/io/snappydata/impl/SnappyHiveCatalog.java

```diff
@@ -375,11 +375,26 @@ private void initHMC() {
       DriverRegistry.register("io.snappydata.jdbc.EmbeddedDriver");
       DriverRegistry.register("io.snappydata.jdbc.ClientDriver");
 
-      String url = "jdbc:snappydata:;user=" +
-          SnappyStoreHiveCatalog.HIVE_METASTORE() +
-          ";disable-streaming=true;default-persistent=true";
       HiveConf metadataConf = new HiveConf();
-      metadataConf.setVar(HiveConf.ConfVars.METASTORECONNECTURLKEY, url);
+      String urlSecure = "jdbc:snappydata:" +
+          ";user=" + SnappyStoreHiveCatalog.HIVE_METASTORE() +
+          ";disable-streaming=true;default-persistent=true";
+      final Map<Object, Object> bootProperties = Misc.getMemStore().getBootProperties();
+      if (bootProperties.containsKey("user") && bootProperties.containsKey("password")) {
+        urlSecure = "jdbc:snappydata:" +
+            ";default-schema=" + SnappyStoreHiveCatalog.HIVE_METASTORE() +
+            ";user=" + bootProperties.get("user") +
+            ";password=" + bootProperties.get("password") +
+            ";disable-streaming=true;default-persistent=true";
+        metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME,
+            bootProperties.get("user").toString());
+        metadataConf.setVar(HiveConf.ConfVars.METASTOREPWD,
+            bootProperties.get("password").toString());
```

Comment:
Do you need the properties both in URL and in metadataConf, or only latter should work?